syntaxes/test/data/expression.html

```diff
@@ -0,0 +1,80 @@
+<!-- Data types -->
+{{ variable }}
+{{ 'string' }}
+{{ "string" }}
+{{ 1 }}
+{{ true }}
+{{ false }}
+{{ null }}
+{{ this }}
+{{ undefined }}
+{{ [variable, variable] }}
+{{ ['string', 'string'] }}
+{{ ["string", "string"] }}
+{{ [1, 2] }}
+{{ [true, false, null, this, undefined] }}
+{{ {object: variable} }}
+{{ {object: 'string'} }}
+{{ {object: "string"} }}
+{{ {object: 1} }}
+{{ {object: true} }}
+{{ {object: false} }}
+{{ {object: null} }}
+{{ {object: this} }}
+{{ {object: undefined} }}
+{{ {object: [variable, variable]} }}
+{{ {object: ['string', 'string']} }}
+{{ {object: ["string", "string"]} }}
+{{ {object: [1, 2]} }}
+{{ {object: [true, false, null, this, undefined]} }}
+
+<!-- Function call -->
+{{ call() }}
+{{ call(); }}
+{{ call().object }}
+{{ call().object; }}
+{{ call()?.conditional }}
+{{ call()?.conditional; }}
+{{ call()!.conditional }}
+{{ call()!.conditional; }}
+
+<!-- Nested function call -->
+{{ object.call() }}
+{{ object.call(); }}
+{{ object.call().object }}
+{{ object.call().object; }}
+{{ object?.call()?.conditional }}
+{{ object?.call()?.conditional; }}
+{{ object!.call()!.conditional }}
+{{ object!.call()!.conditional; }}
+
+<!-- Function parameter test -->
+{{ call(withParams) }}
+{{ call('withParams') }}
+{{ call("withParams") }}
+{{ call("withParams") }}
+{{ call(false) }}
+{{ call(!!true) }}
+{{ call(["array", 123]) }}
+{{ call(1) }}
+{{ call({ test: 'object' }) }}
+{{ call({ test: 123 }) }}
+{{ call({ test: [123, 321, { value: ['string', false, true] }] }) }}
+
+<!-- Ternary expression -->
+{{ condition ? true : false }}
+{{ condition ? true : false; }}
+{{ condition() ? call(1 + 2 + 3) : call()  }}
+{{ condition() ? call(1 + 2 + 3) : call();  }}
+{{ condition()?.object ? call()!.test() : false }}
+{{ condition ? ['123'] : { test: 'a' } }}
+{{ condition ? 'test' : "test" }}
+{{ condition ? [function(), variable] : {} }}
+
+<!-- Expression test -->
+{{ (let param of params?.anotherParam!.param().param | async) as p; let i = index; let first = first }}
```

Comment:
Let's make this a little more atomic. It's not clear what's being tested here..

Reply:
I added a specific part on the tests as "mixed test" to test the microsyntax against the expression